src/lpython/parser/parser.cpp

```diff
@@ -42,6 +42,33 @@ Result<LPython::AST::Module_t*> parse(Allocator &al, const std::string &s,
         p.result.p, p.result.size(), p.type_ignore.p, p.type_ignore.size());
 }
 
+Result<LPython::AST::ast_t*> parse_to_ast(Allocator &al, const std::string &s,
+        uint32_t prev_loc, diag::Diagnostics &diagnostics)
+{
+    Parser p(al, diagnostics);
+    try {
+        p.parse(s, prev_loc);
+    } catch (const parser_local::TokenizerError &e) {
+        Error error;
+        diagnostics.diagnostics.push_back(e.d);
+        return error;
+    } catch (const parser_local::ParserError &e) {
+        Error error;
+        diagnostics.diagnostics.push_back(e.d);
+        return error;
+    }
+
+    Location l;
+    if (p.result.size() == 0) {
+        l.first=0;
+        l.last=0;
+    } else {
+        l.first=p.result[0]->base.loc.first;
+        l.last=p.result[p.result.size()-1]->base.loc.last;
+    }
+    return LPython::AST::make_Interactive_t(al, l, p.result.p, p.result.size());
```

Comment:
Any specific reason why we use `make_Interactive_t` here? It seems if we reuse the existing `parse()` function (on line `17` of this file) which returns a `Module` we can save a lot of code being added in this PR.

For example, almost all the code added in `python_ast_to_asr.cpp` would not be needed. It seems to be duplicate of existing code for `visit_Module()`.

Reply:
There are some differences between `SymbolTreeVisitor.visit_Interactive` and `SymbolTreeVisitor.visit_Module`.
I will try merging both. Most probably it should be possible. Then we can remove `make_Interactive_t` and reuse `visit_Module` at the places `visit_Interactive` is used.